Update jetpack-connection-banner to Emerald style

robertsreberski opened the pull request:
This PR introduces revamped Jetpack Connection banner to match the recently introduced Emerald style. The new banner layout is strongly based on the current Jetpack Recommendations banner implementation.

Additionally, this PR cleans up banner styles and scripts that are not used anymore. It aims to declutter the banners' logic and facilitate the creation of any other banners in Emerald style in the future.

#### Changes proposed in this Pull Request:
- Update the Jetpack Connection banner to the Emerald style (same as the Recommendations banner)
- Clean up banners styles and scripts that are no longer used

#### Jetpack product discussion
p6TEKc-6dA-p2

#### Testing instructions:

##### Test a default banner
* Checkout this branch `git checkout update/jetpack-connection-banner-style`
* Run the WordPress local env with the jetpack plugin
* Run `jetpack watch` and select `plugins > jetpack` in order to build the new code introduced in this PR
* Ensure you're disconnected from WordPress in your current env
* Go to the dashboard `/wp-admin/index.php`
* You should see a new banner:
<img width="1490" alt="image" src="https://user-images.githubusercontent.com/8419292/170684705-fe0fb428-4a3c-4b2f-b779-917715095dfb.png">

* Go to plugins page `/wp-admin/plugins.php`
* You should see the same banner as well:
<img width="1714" alt="image" src="https://user-images.githubusercontent.com/8419292/170684805-59412d24-8fb2-4857-9835-87fc8c3d7016.png">

* Clicking "Set up Jetpack" should connect you to Jetpack; the banner should not display anymore after that point

##### Test a license aware banner
* Disconnect from Jetpack
* Replace the line 149 with `add_action( 'admin_notices', array( $this, 'render_license_aware_banner' ) );` to force license aware banner to display
* Go to the dashboard `/wp-admin/index.php`
* You should see the banner in Emerald style with a different copy attached:
<img width="1713" alt="image" src="https://user-images.githubusercontent.com/8419292/170686838-de84a242-beae-42a1-8ba1-1c001b5bdb38.png">

#### Does this pull request change what data or activity we track or use?

No.

robertsreberski committed:
Replace assets for the new banner style

robertsreberski committed:
Unify emerald styled banners

robertsreberski committed:
Clean up unused banner styles and scripts

robertsreberski committed:
Update connect-button scripts to the new emerald banner

robertsreberski committed:
Add jetpack-connection-banner emerald styles

robertsreberski committed:
Add jetpack-connection-banner emerald layout and copy

robertsreberski committed:
changelog

robertsreberski committed:
Update version in function docs

robertsreberski committed:
Add new copy to license-aware version of banner

robertsreberski committed:
Merge branch 'master' into update/jetpack-connection-banner-style

monsieur-z reviewed (changes requested):
The PR tested well. I think the design needs some adjustments on mobile, like the CTA text being centered.
<img width="355" alt="Screen Shot 2022-05-30 at 2 13 43 PM" src="https://user-images.githubusercontent.com/1620183/171047519-636d3f6f-0007-45da-b4df-600d2292112a.png">
<img width="358" alt="Screen Shot 2022-05-30 at 2 44 45 PM" src="https://user-images.githubusercontent.com/1620183/171047522-74c78cde-c04b-4328-809e-41ef456f579b.png">

I'm not familiar with that part of the code, so it's difficult for me to say if you removing those lines could have an impact or not. If you're sure it's okay, go with it.

There's at least 1 e2e test that this PR is breaking.

monsieur-z commented on projects/plugins/jetpack/class.jetpack-connection-banner.php:
```diff
@@ -251,91 +267,60 @@ public static function enqueue_connect_button_scripts() {
+							srcset="<?php echo esc_url( plugins_url( 'images/jetpack-connection-image.webp', JETPACK__PLUGIN_FILE ) ); ?> 1x, <?php echo esc_url( plugins_url( 'images/jetpack-connection-image-2x.webp', JETPACK__PLUGIN_FILE ) ); ?> 2x">
+						<img
+							class="jp-connection-banner__image"
+							srcset="<?php echo esc_url( plugins_url( 'images/jetpack-connection-image-2x.png', JETPACK__PLUGIN_FILE ) ); ?> 2x"
```
Not 100% sure, but I think you may need to add a 1x version in `srcset` too for users with a browser that might support it, but with a low res screen.

monsieur-z commented on projects/plugins/jetpack/scss/jetpack-connection-banner.scss:
```diff
@@ -0,0 +1,161 @@
+@import 'jetpack-banners-emerald';
+
+$jp-connection-banner-breakpoint: 1152px;
```
Is this breakpoint arbitrary, or used elsewhere in the plugin?

robertsreberski committed:
Add 1x asset to img srcset

robertsreberski committed:
Center CTA text on mobile

robertsreberski committed:
Update e2e test to the new banner classes

robertsreberski committed:
Unify md breakpoint for emerald banners

robertsreberski commented:
Thanks for the review @monsieur-z! I've updated the code based on your suggestions! Thanks for pointing out the failing e2e test - I've addressed that issue too.

> I'm not familiar with that part of the code, so it's difficult for me to say if you removing those lines could have an impact or not. If you're sure it's okay, go with it.

I've statically checked the code to find usages in the repository of CSS classes I've decided to remove. I couldn't find any indications that they are still in use. I don't see any broken elements within the Jetpack plugin code. While I understand it's "a risky move", I think it's worth it, especially during the moment we do transition to the new styles. Otherwise, we will end up with a lot of code supporting both new and legacy versions of banners that add complexity to the project.

robertsreberski committed:
Bump version to 11.1 in docs and tracking

robertsreberski committed:
Merge branch 'master' into update/jetpack-connection-banner-style

robertsreberski committed:
Merge branch 'master' into update/jetpack-connection-banner-style

samiff committed:
Merge branch 'master' into update/jetpack-connection-banner-style

samiff committed:
Merge branch 'trunk' into update/jetpack-connection-banner-style

samiff commented on projects/plugins/jetpack/class.jetpack-connection-banner.php:
```diff
@@ -251,91 +267,60 @@ public static function enqueue_connect_button_scripts() {
-	 * @since 7.2   Copy and visual elements reduced to show the new focus of Jetpack on Security and Performance.
 	 * @since 4.4.0
+	 * @since 7.2   Copy and visual elements reduced to show the new focus of Jetpack on Security and Performance.
+	 * @since 11.1  Adjusted the banner to Emerald style
```
```suggestion
	 * @since $$next-version$$  Adjusted the banner to Emerald style
```

You can use `$$next-version$$` in cases like this which will be automatically replaced with the correct package/plugin version number 👍

samiff commented on projects/plugins/jetpack/class.jetpack-connection-banner.php:
```diff
@@ -346,64 +331,46 @@ class="dops-button is-primary jp-banner__alt-connect-button">
 	 * Renders the license-away version of the connection banner.
 	 *
 	 * @since 9.0.0
+	 * @since 11.1  Adjusted the banner to Emerald style
```
```suggestion
	 * @since $$next-version$$  Adjusted the banner to Emerald style
```

robertsreberski replied:
Thanks, will remember this in the future!

samiff commented on projects/plugins/jetpack/class.jetpack-connection-banner.php:
```diff
@@ -251,91 +267,60 @@ public static function enqueue_connect_button_scripts() {
+					<div class="jp-connection-banner__logo">
+						<?php echo $jetpack_logo->get_jp_emblem_larger(); // phpcs:ignore WordPress.Security.EscapeOutput.OutputNotEscaped ?>
+					</div>
+					<h2 class="jp-connection-banner__title"><?php esc_html_e( 'Simplify your site security and performance with&nbsp;Jetpack', 'jetpack' ); ?></h2>
```
```suggestion
					<h2 class="jp-connection-banner__title"><?php esc_html_e( 'Simplify your site security and performance with Jetpack', 'jetpack' ); ?></h2>
```

Is there a particular reason for using `&nbsp;` here?

robertsreberski replied:
I just wanted to avoid [text widows](https://en.wikipedia.org/wiki/Widows_and_orphans) that happen in specific edge cases, but indeed it's not a good option to put it in the translation. And we cannot really avoid widows with the translatable text.

samiff commented on projects/plugins/jetpack/class.jetpack-connection-banner.php:
```diff
@@ -346,64 +331,46 @@ class="dops-button is-primary jp-banner__alt-connect-button">
+						</svg>
+						<?php esc_html_e( 'Finish setting up Jetpack', 'jetpack' ); ?>
+					</h2>
+					<div class="jp-connection-banner__text"><?php esc_html_e( "Thanks for purchasing a Jetpack subscription.\nThere’s just one more step to complete the installation.", 'jetpack' ); ?></div>
```
@jeherve or @anomiex Do you see any issues with using `\n` in this `esc_html_e()` call? It seems "weird" to me to have escaped characters like that in a translated string but maybe it's fine 🤷

anomiex replied:
It'll work fine. Unless they're doing something with CSS `white-space` or something, though, it won't display any differently compared to just using a space.

robertsreberski replied:
I confirm there is a [`white-space` property definition in this class](https://github.com/Automattic/jetpack/pull/24501/files#diff-bbe9eff3dd4ae5331cab559a9bcccbc8bf125b6c162d337b1b4b39e3ab453860R99) that handles a new line.

robertsreberski committed:
Update projects/plugins/jetpack/class.jetpack-connection-banner.php

Co-authored-by: Samiff <samiff@users.noreply.github.com>

robertsreberski committed:
Update projects/plugins/jetpack/class.jetpack-connection-banner.php

Co-authored-by: Samiff <samiff@users.noreply.github.com>

robertsreberski committed:
Update projects/plugins/jetpack/class.jetpack-connection-banner.php

Co-authored-by: Samiff <samiff@users.noreply.github.com>

samiff reviewed (approved):
Tested the different views (including license) on desktop/mobile and looked good to me 👍
